CIP-0030/README.md

```diff
@@ -194,15 +234,60 @@ In order to initiate communication from webpages to a user's Cardano wallet, the
 
 Errors: APIError
 
-This is the entrypoint to start communication with the user's wallet. The wallet should request the user's permission to connect the web page to the user's wallet, and if permission has been granted, the full API will be returned to the dApp to use. The wallet can choose to maintain a whitelist to not necessarily ask the user's permission every time access is requested, but this behavior is up to the wallet and should be transparent to web pages using this API. If a wallet is already connected this function should not request access a second time, and instead just return the `API` object.
+This is the entrypoint to start communication with the user's wallet.
+The wallet should request the user's permission to connect the dApp to the user's wallet.
+Only if permission has been granted, will an API object will be returned to the dApp to use.
+
+The wallet can choose to maintain a dApp whitelist to minimize asking for the user's permission every time access is requested.
+However, this behavior is wallet specific and should be transparent to dApps using this API.
+
+If this method is called with a new extension list, and the dApp is asking for a new extension that would require specific user consent to activate, then it should be treated as if it was called for the first time.
+This is to prevent a possibility where a dApp asks for basic extensions and then asks for an extension that the wallet would otherwise have prompted the user for permission to enable.
+Otherwise, if the extension list has not changed OR the wallet does not ask for permissions to enable any individual extensions, then this function should not request access a second time, and instead, just return the `API` object.
+
+Upon start, dApp can explicitly request a list of additional functionalities they expect as a list of CIP numbers capturing those extensions.
+This is used as an extensibility mechanism to document what functionalities can be provided by the wallet interface.
+CIP-0030 provides a set of base interfaces that every wallet must support.
+Then, new functionalities are introduced via additional CIPs and may be all or partially supported by wallets.
+
+If the extensions list is not specified, it defaults to an empty list `{}`.
+The wallet should treat an empty list as a request to enable **ALL** extensions which are not mutually exclusive.
```

Comment:
By enabling all wallets do lose the ability to permission per extension.

Although nothing was stopping wallets from enabling all supported extensions regardless of what dApps were asking for - this would make wallet implementations simpler. Additionally, I am unsure of the real utility of permissioning access to extensions - I cannot think of a usecase.

Reply:
There is no security implication documented here in CIP-30 for enabling extensions.  Nor how such a flow would work.
There is also no signalling back to the dApp that it's security restricted vs just not available.
For example,  if a phone app asks to use the camera, it's told either "there is no camera" OR "you don't have permission to use the camera", which are different states to which the app can make different UX decisions.
If this is intended to be a security mechanism, then that should have proper consideration and specification.